File: app/src/main/java/buddybuild/com/ultron/MyApplication.java
package buddybuild.com.ultron;

import android.app.Activity;
import android.app.Application;

import javax.inject.Inject;

import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasDispatchingActivityInjector;

public class MyApplication extends Application implements HasDispatchingActivityInjector{
    @Inject
    DispatchingAndroidInjector<Activity> dispatchingAndroidInjector;

    private static IApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        applicationComponent = DaggerIApplicationComponent.builder().build();
    }

    public static IApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }

    @Override
    public DispatchingAndroidInjector<Activity> activityInjector() {
        return dispatchingAndroidInjector;
    }
}

File: app/src/main/java/buddybuild/com/ultron/MyApplication.kt
package buddybuild.com.ultron

import android.app.Application

class MyApplication : Application() {

    companion object {
        lateinit var appComponent: ApplicationComponent
    }
    override fun onCreate() {
        super.onCreate()

        appComponent = DaggerApplicationComponent.builder()
                .activityModule(ActivityModule)
                .build()
    }

}
